common/lib/capa/capa/capa_problem.py

```diff
@@ -798,16 +798,20 @@ def _extract_html(self, problemtree):  # private
             if problemid in self.correct_map:
                 pid = input_id
 
-                # If the the problem has not been saved since the last submit set the status to the
-                # current correctness value and set the message as expected. Otherwise we do not want to
-                # display correctness because the answer may have changed since the problem was graded.
-                if not self.has_saved_answers:
-                    status = self.correct_map.get_correctness(pid)
-                    msg = self.correct_map.get_msg(pid)
-
-                hint = self.correct_map.get_hint(pid)
-                hintmode = self.correct_map.get_hintmode(pid)
-                answervariable = self.correct_map.get_property(pid, 'answervariable')
+                # If we're withholding correctness, don't show hints either.
```

Comment:
Thanks for the clarification, @cahrens .  34fbb5f9bd adds "adaptive" to the existing comment, and adds a line about the "demand" hints.